flatpak-proxy.c

```diff
@@ -2959,7 +2982,48 @@ side_in_cb (GSocket *socket, GIOCondition condition, gpointer user_data)
               got_buffer_from_side (side, buffer);
 
               if (found_auth_end)
-                client->authenticated = TRUE;
+                client->auth_state = client->auth_server_backlog
+                  ? AUTH_WAITING_FOR_BACKLOG
+                  : AUTH_COMPLETE;
```

Comment:
Could use some braces.

Reply:
And maybe be more explicit: `client->auth_server_backlog > 0`

Reply:
Mmm... This is the original commit from @refi64, which I based my work on. I included in the branch as-is and then I added my fixes on top of his.

I don't think I should change this commit. Maybe then it's better if I merge his and my commit (and I will fix all these remarks), and then I just credit him in the commit message? WDYT?